packages/cursorless-vscode/src/ide/vscode/VscodeIDE.ts

```diff
@@ -126,6 +126,14 @@ export class VscodeIDE implements IDE {
     return this.fromVscodeEditor(await window.showTextDocument(textDocument));
   }
 
+  public async openUntitledTextDocument(options?: {
+    language?: string;
+    content?: string;
+  }): Promise<TextEditor> {
```

Comment:
I'd import and use type `OpenUntitledTextDocumentOptions` from our `ide.types.ts`

Reply:
Yes missed that one. You like the name `openUntitledTextDocument`? Vscode just has a overloaded function that can take either a uri or these options.

Reply:
Maybe `newUntitledTextEditor`? I wouldn't say I strongly prefer your api or VSCode's, but I agree it's a bit cumbersome with their api that the client needs to call two functions

Reply:
Yet that part is annoying, but what I was getting at is if we want to have two separate functions. `openTextDocument` and `openUntitledTextDocument` or do what vscode does and just have the single `openTextDocument` that does both depending on passed arguments?

Reply:
I would vote for two separate functions; overloading is hard to read at the call site. (They can both dispatch to a shared, overloaded function under the hood, if desired.)

Reply:
Ok then I guess we leave it as is :)

Reply:
In terms of function name, I think I'd prefer `newUntitledTextEditor` or `openNewUntitledTextEditor`, to indicate it's returning an editor, not a document

Reply:
The same is true for `openTextDocument` I think we should be consistent in the naming of these two.